### PR TITLE
Intellij IDEA Community Edition

### PR DESCRIPTION
**Pull request by kevincianfarini**

### CODE REVIEW

File: com.jetbrains.Intellij-Community.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2015 Petr Hracek <phracek@redhat.com> -->
+<application>
+    <id type="desktop">com.jetbrains.Intellij-Community.desktop</id>
+    <licence>CC-BY-SA</licence>
```
Comment (Alexander-Wilms) [inline com.jetbrains.Intellij-Community.appdata.xml]: I think this should be

<project_license>Apache-2.0</project_license>
<metadata_license>CC-BY-SA-4.0</metadata_license> (or 3.0, 2.0, 1.0, cf. https://spdx.org/licenses/)

cf.  https://www.jetbrains.com/idea/download/#section=linux

**Comment by nedrichards** [comment]
bot, build com.jetbrains.Intellij-Community

**Comment by kevincianfarini** [comment]
I can change the license. I was modelling this off of the pycharm flatpak and had just assumed that licensing would be the same for both. 
Also, how long does the bot take to complete a build?

**Comment by nedrichards** [comment]
bot, build com.jetbrains.Intellij-Community

**Comment by nedrichards** [comment]
Link here: https://flathub.org/builds/#/builders/2/builds/146 (updating github is a bit unreliable right now)

**Comment by kevincianfarini** [comment]
Is there anything else I should do before this is eligible for a merge?

**Comment by nedrichards** [comment]
@kevincianfarini how does it feel to use? As a developer, is anything particularly useful or obvious missing? Are there any java tools that should be required that we need to bundle?

These kinds of IDEs have users who like to stretch the boundaries (in a good way) so it's good to make sure we're covering most things.

**Comment by kevincianfarini** [comment]
@nedrichards as far as I know, no. I use IntelliJ Community and all plugins (which includes language and framework support) for the project are provided from Jetbrains. For all of the use cases for both Pycharm and IntelliJ I've ever had, things were bundled with the IDE or available for download.

**Comment by adityashah1212** [comment]
I may be wrong but I think IntelliJ requires you to give path for any build systems other than ANT.

**Comment by kevincianfarini** [comment]
@adityashah1212 Could you elaborate? This isn't building from IJ-C source if you're under that impression. It's downloading the build from their release page.

**Comment by adityashah1212** [comment]
No I mean build systems such as maven and gradle generally used by java projects.

**Comment by nedrichards** [comment]
Thanks @adityashah1212 that's the kind of thing I mean. Shipping a copy of gradle (if required for example) and making sure IntelliJ knows where it is will significantly reduce user frustration and issues post release.

**Comment by kevincianfarini** [comment]
Just out of curiosity, I deleted my local non-flatpak installations on IntelliJ and reinstalled my personal flatpak build. I'm not having problems with gradle and maven at all on my projects. Can you give an example of providing PATH to IntelliJ because I don't know where to look into this. 

That being said, the installation ran fine but didn't recognize my system JDK version of 1.8, but did see the JDK 9 that's bundled with the flatpak. Also I had to manually change the integrated terminal to `/bin/bash` from `/bin/sh` in the IDE settings, which isn't normal I don't believe. Any ideas what that's about? I tried looking at the Android Studio repo for guidance but didn't see anything of interest. 

@adityashah1212 @nedrichards

**Comment by adityashah1212** [comment]
Hmm... Not sure of maven and gradle, but I am sure you need to allow access to the host filesystem on flatpak, as far as how to get that jdk to show up in your IntelliJ, I think it is better for the user to manually add it to IntelliJ settings. And as to why it is choosing /bin/sh over /bin/bash, I can only speculate that, IntelliJ is somehow trying to detect the system it is running on and not able to recognize it or something on those lines.

**Comment by ghisvail** [comment]
Are there any blockers currently?

**Comment by barthalion** [comment]
Is maven, gradle and similar tools bundled with IntelliJ or needs to be added separately?

**Comment by x80486** [comment]
As far as I know, `Maven` is the only build tool bundled with **IntelliJ IDEA**, and by default `Maven` stores everything in `$HOME/.m2`, so I think the user's home directory should be accesible by default.

On a side note, is this blocked by anything?

**Comment by amtlib-dot-dll** [comment]
Perhaps newer ones like `org.freedesktop.Sdk/x86_64/18.08` as well as `org.freedesktop.Sdk.Extension.openjdk11/x86_64/18.08` should be used instead of the 1.6 versions. BTW, `fs-notifier` should not be removed as per https://github.com/flathub/com.jetbrains.PyCharm-Community/pull/27.

Oracle says that the Oracle JDK and the OpenJDK are "Functionally identical and interchangeable" here https://blogs.oracle.com/java-platform-group/oracle-jdk-releases-for-java-11-and-later, since JDK 11

> by default `Maven` stores everything in `$HOME/.m2`, so I think the user's home directory should be accesible by default.

Not necessarily, `--persist=.m2` is also an option

File: com.jetbrains.Intellij-Community.json [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+              "build-commands": [
+                "/usr/lib/sdk/openjdk9/install.sh"
+              ]
+          },
+          {
+            "name": "intellij",
+            "buildsystem": "simple",
+            "build-commands": [
+                "tar xzfv ideaIC-2017.3.5.tar.gz",
+                "rm -f ideaIC-2017.3.5/idea-IC-173.4674.33/bin/fsnotifier*",
```
Comment (amtlib-dot-dll) [inline com.jetbrains.Intellij-Community.json]:
```suggestion

```
Probably `fs-notifier` should be kept as explained in https://github.com/flathub/com.jetbrains.PyCharm-Community/pull/27

**Comment by amtlib-dot-dll** [comment]
Additionally, there are two versions of compiled IntelliJ IDEA, one [with JDK](_https://www.jetbrains.com/idea/download/download-thanks.html?platform=linux&code=IIC) and another [without](_https://www.jetbrains.com/idea/download/download-thanks.html?platform=linuxWithoutJDK&code=IIC). Why bother bundle one? 😂

**Comment by x80486** [comment]
Bundling the Java SDK with IntelliJ IDEA goes beyond that. It really makes no sense because then we would need to bundle all available (and possible not available) versions since we don't know what version the final user is going to use. The same with the rest of the SDKs.

I do not see a need to bundle any. Users can use their own ones installed _somewhere_, provided that _somewhere_ is accessible to the Flatpak.

**Comment by amtlib-dot-dll** [comment]
Yes, they are accessible under `/run/host/usr` https://github.com/flathub/com.visualstudio.code.oss/issues/22#issuecomment-433629703

Additionally, if users manually install any one of `org.freedesktop.Sdk.Extension.openjdk*`, these JDKs are also accessible and usable.

**Comment by barthalion** [comment]
Using anything from `/run/host/usr` is asking for trouble. You can't assume it will work with the runtime.

**Comment by x80486** [comment]
Indeed, @barthalion, I've faced that before while testing the Ultimate Edition of IDEA.

Is it possible for a Flatpak that doesn't bundle a given `Sdk.Extension` –for a given SDK-, to be able to access it without bundling it/them in the original Flatpak? If so, problem solved and we just need to have different versions of those...although the approach to install the SDKs should be stated somewhere because it might be a little bit confusing to some folks starting using those artifacts/Flatpaks in general.

**Comment by amtlib-dot-dll** [comment]
@x80486 Yes. You can try it. 

@barthalion It might work, and that's enough, at least for now. Actually in some cases it won't work like here https://github.com/flathub/com.visualstudio.code.oss/issues/22#issuecomment-433705062

**Comment by x80486** [comment]
Hi @kevincianfarini, are you still around? Is it possible we can move forward with this one?

**Comment by x80486** [comment]
> Additionally, if users manually install any one of `org.freedesktop.Sdk.Extension.openjdk*`, these JDKs are also accessible and usable.

@amtlib-dot-dll, I installed the `OpenJDK 11.x` SDK extension individually in order to test this. I also installed IntelliJ IDEA (as a Flatpak)...but there is no way I can see where the location for the JDK is when I'm trying to reference it from within the IDE in order to "register" it:

```shell
$ flatpak list
Ref                                                  Options       
com.jetbrains.IntelliJ-IDEA-Ultimate/x86_64/master   user,current  
org.freedesktop.Sdk.Extension.openjdk11/x86_64/18.08 user,runtime  
org.freedesktop.Sdk.Extension.openjdk11/x86_64/18.08 system,runtime
...
org.freedesktop.Sdk/x86_64/1.6                       system,runtime
```

screenshot from 2018-11-10 17-31-46

**Comment by nedrichards** [comment]
Merged in #764

**Pull request closed without merging**